src/app/(auth)/login/page.tsx

```diff
@@ -42,26 +46,29 @@ export default function Login() {
       return;
     }
 
-    const { error } = await auth.signIn(email, password);
+    setIsLoggingIn(true);
+    const { error, data } = await auth.signIn(email, password);
 
     if (error) {
       setErrorMessage(error.message);
       // TODO: use error.status to check if it's an email-specific or password-specific error
       // then, raise the error in the TextInput component.
     } else {
-      profile?.loadProfile();
       setErrorMessage('');
-      setFinishLogin(true);
+      const loadedProfile = await profile.loadProfile(data.user?.id);
 
-      // conditional routing after logging in
-      if (profile?.profileReady) {
-        if (!profile?.profileData) push(CONFIG.onboardingHome);
-        else if (profile.roles.map(r => r.role).includes('ATTORNEY'))
-          push(CONFIG.cases);
-        else if (profile.roles.map(r => r.role).includes('LEGAL_FELLOW'))
-          push(CONFIG.lca);
-        else push(CONFIG.languageSupport);
+      if (!loadedProfile) {
+        push(CONFIG.onboardingHome);
+        return;
       }
+
+      // conditional routing after logging in
+      if (!loadedProfile.profileData) push(CONFIG.onboardingHome);
```

Comment:
can we combine these two conditions? like: 

`if (!loadedProfile || !loadedProfile.profileData) push(CONFIG.onboardingHome); `

Reply:
we could, but then we'd also have to transform every other check below it to also include `!loadedProfile`

the reason for adding the first `if (!loadedProfile)` check is to just ensure it won't be null - if we join this check with `!loadedProfile.profileData`, then we'd also have to do this for the checks below, since it can be null

both would work, but i think this ultimately reduces the number of boolean checks required

Reply:
will there ever be a case where loadedProfile is truthy, but loadedProfile.profileData) is falsy? 
if not, wouldn't combining the two checks be fine w/o rechecking for every other check?